hikvision_tftpd.py

```diff
@@ -73,6 +73,17 @@ def _bind(self, addr):
             raise
         return sock
 
+    def set_block_size(self, block_size):
```

Comment:
It seems a little odd to mutate the whole server for one transfer. Before, it was so dumb as to be totally stateless. Now, it has transfer-specific state but stores it as a singleton, so there are a couple cases it wouldn't handle well:

* simultaneous transfers with different options. It will use the options of whatever started latest.

* a transfer with options, then a transfer without options. It will keep the original options.

I think the "right" thing would be to add a notion of per-transfer state. But I suppose with the dumb way the Hikvision devices all try to use the same IP, you're probably only going to be using it with one device at a time anyway, so it's probably fine. We could add a comment (TODO / caveat / whatever you want to call it) instead if you prefer.

Reply:
I 100% agree and this is something I thought a lot about. For what we need the server for the assumptions that only one client will connect at a time and that once the transfer is complete the server will be shut down were able to be made. I decided that if I was building a real TFTP server I would fix this however for the exact purposes of this hikvision specific recovery mode it felt like a lot of extra work to do.